libs/utils/decorate.js

```diff
@@ -0,0 +1,129 @@
+    Object.entries(icon).forEach((value) => {
+      const iconValue = value[1];
+      if (iconValue) {
+        iconValues[value[0]] = iconValue;
+      }
+    });
+    library[icon.key] = iconValues;
+  });
+  return library;
+}
+
+export async function decorateIcons(tokenLibrary) {
+  const iconLibrary = await formatIconLibraryData(tokenLibrary);
+  const el = document.querySelector('main');
+  const icons = el.querySelectorAll('.icon');
+  icons?.forEach((i) => {
+    const str = i.textContent;
+    const icon = iconLibrary[str];
+    const size = str.includes('persona') ? 80 : 40;
+    if (iconLibrary && icon) {
+      const styles = icon.key.replaceAll('-', ', ').split(', ');
+      if (styles) i.classList.add(icon.key, ...styles);
+      i.classList.add(icon.key)
+      const svg = `<img height="${size}" width="${size}" alt="${icon.label}" src="${icon.value}">`;
+      const label = `${svg} ${(icon.label !== undefined) ? icon.label : ''}`;
+      const anchor = `<a class="icon ${str}" href="${icon.link}">${label}</a>`;
+      if (icon.link !== undefined) {
+        i.outerHTML = anchor;
+      } else {
+        i.innerHTML = i.innerHTML.replace(str, label);
+      }
+    }
+  });
+}
+
+export function decorateBlockText(el, size = 'small') {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  const decorate = (headingEl, headingSize, bodySize, detailSize) => {
+    headingEl.classList.add(`heading-${headingSize}`);
+    headingEl.nextElementSibling.classList.add(`body-${bodySize}`);
+    if (headingEl.previousElementSibling) {
+      headingEl.previousElementSibling.classList.add(`detail-${detailSize}`);
+    }
+  };
+  if (size === 'medium') {
+    decorate(heading, 'M', 'S', 'M');
+  } else if (size === 'large') {
+    decorate(heading, 'XL', 'M', 'L');
+  } else {
+    decorate(heading, 'XS', 'S', 'M');
+  }
+  initIcons(el);
+  decorateButtons(el);
+  decorateLinkAnalytics(el, heading);
+}
+
+export function isHexColorDark(color) {
+  if (!color.trim().startsWith('#')) return false;
+  const hex = color.trim().replace('#', '');
+  const cR = parseInt(hex.substr(0, 2), 16);
+  const cG = parseInt(hex.substr(2, 2), 16);
+  const cB = parseInt(hex.substr(4, 2), 16);
+  const brightness = ((cR * 299) + (cG * 587) + (cB * 114)) / 1000;
+  return brightness < 155;
+}
+
+export function decorateBlockBg(block, node) {
+  node.classList.add('background');
+  if (!node.querySelector(':scope img')) {
+    block.style.background = node.textContent;
+    if (isHexColorDark(node.textContent)) block.classList.add('dark');
+    node.remove();
+  }
+}
+
+export function getBlockSize(el) {
+  const sizes = ['small', 'medium', 'large'];
+  return sizes.find((size) => el.classList.contains(size)) || sizes[1];
+}
+
+export async function getTokenLibrary(path = '/docs/library/tokens.json') {
+  const url = (window.location.port === '2000') ? `https://main--milo--adobecom.hlx.page${path}`
```

Comment:
The tests should not depend on a network connection being available.  You can mock tokens.json for the test. (similar to https://github.com/adobecom/milo/blob/main/test/libs/blocks/caas-config/mockFetch.js)

Reply:
@chrischrischris - I refactored this so we can pass fetchURL in testing and removed this network connection need. LMK if we need to tweak this. Looking good w/ 100% coverage now.